### PR TITLE
feat(command): add force flag for files rm

### PR DESCRIPTION
**Pull request by overbool**
Refer to: https://github.com/ipfs/go-ipfs/issues/5538#issuecomment-426312110, https://github.com/ipfs/go-ipfs/issues/5554

License: MIT
Signed-off-by: Overbool <overbool.xu@gmail.com>

### CODE REVIEW

File: core/commands/files.go [hunk 1]
```diff
@@ -1002,6 +1002,7 @@ Remove files or directories.
 	},
 	Options: []cmdkit.Option{
 		cmdkit.BoolOption("recursive", "r", "Recursively remove directories."),
+		cmdkit.BoolOption("force", "Allow to remove anything else"),
```
Comment (eingenito) [inline core/commands/files.go]: What do you think of `forcibly remove target at path; implies -r for directories` as a description?

**Comment by overbool** [comment]
@schomatis could u help me review this pr?

**Comment by schomatis** [comment]
The `-f` short flag is already used for the global flush flag,

https://github.com/ipfs/go-ipfs/blob/a86cde50de4bd8da0b5eaf6f1936269bafd29c8e/core/commands/files.go#L56,

but that's ok, I actually prefer to just have the long explicit `--force` flag since this should be used with care and (hopefully) not too often.

File: core/commands/files.go [hunk 2]
```diff
@@ -1066,25 +1067,31 @@ Remove files or directories.
...
 		if err != nil {
 			res.SetError(err, cmdkit.ErrNormal)
 			return
 		}
 
-		switch childi.(type) {
+		dashr, _, _ := req.Option("r").Bool()
+
+		switch child.(type) {
 		case *mfs.Directory:
```
Comment (schomatis) [inline core/commands/files.go]: nit: we can now turn this single-case switch into an `if`

File: test/sharness/t0250-files-api.sh [hunk 3]
```diff
@@ -629,6 +629,20 @@ test_files_api() {
   test_expect_success "repo gc $EXTRA" '
     ipfs repo gc
   '
+
+  # test rm
```
Comment (schomatis) [inline test/sharness/t0250-files-api.sh]: I'd like to add a test with an actual corrupted node, which could just be an empty proto-node, but I'm not sure how could we add that to MFS, `ipfs files cp` does check the type:
```
ipfs object new
# QmdfTbBqBPQ7VNxZEYEj14VmRuZBkqFbiwReogJgS1zR1n
ipfs files cp /ipfs/QmdfTbBqBPQ7VNxZEYEj14VmRuZBkqFbiwReogJgS1zR1n /corrupted
# Error: cp: cannot flush the created file /corrupted: proto: required field "Type" not set
```

Comment (schomatis) [reply test/sharness/t0250-files-api.sh]: Oh, turn off flush:

```
ipfs files -f=0 cp /ipfs/QmdfTbBqBPQ7VNxZEYEj14VmRuZBkqFbiwReogJgS1zR1n /corrupted
ipfs files rm /corrupted
Error: proto: required field "Type" not set
```

Comment (schomatis) [reply test/sharness/t0250-files-api.sh]: OT: What's the difference of adding a file in MFS without flushing? I still don't know how that works (and I'm supposedly the one in charge of reviewing the MFS API :grimacing:)

Comment (schomatis) [reply test/sharness/t0250-files-api.sh]: Thinking about this some more, after adding the checks to `PutNode` this won't work anyways (even with flush turned off), so scratch that, I'm fine with the existing tests.
